File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Ram/AlibabaCloud/SDK/ROS/CDK/Ram/Datasource/RolesProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Ram.Datasource
{
    /// <summary>Properties for defining a `Roles`.</summary>
    /// <remarks>
    /// See https://www.alibabacloud.com/help/ros/developer-reference/datasource-ram-roles
    /// </remarks>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-ram.datasource.RolesProps")]
    public class RolesProps : AlibabaCloud.SDK.ROS.CDK.Ram.Datasource.IRolesProps
    {
        private object? _roleName;

        /// <summary>Property roleName: Filter the results by a specific role name.</summary>
        /// <remarks>
        /// Supports using * and ?  to fuzzy match.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "roleName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? RoleName
        {
            get => _roleName;
            set
            {
                if (Amazon.JSII.Runtime.Configuration.RuntimeTypeChecking)
                {
                    switch (value)
                    {
                        case string cast_cd4240:
                            break;
                        case AlibabaCloud.SDK.ROS.CDK.Core.IResolvable cast_cd4240:
                            break;
                        case Amazon.JSII.Runtime.Deputy.AnonymousObject cast_cd4240:
                            // Not enough information to type-check...
                            break;
                        case null:
                            break;
                        default:
                            throw new System.ArgumentException($"Expected {nameof(value)} to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IResolvable).FullName}; received {value.GetType().FullName}", nameof(value));
                    }
                }
                _roleName = value;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkram/datasource/RolesProps.go
package datasource


// Properties for defining a `Roles`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/datasource-ram-roles
type RolesProps struct {
	// Property refreshOptions: The refresh strategy for the datasource resource when the stack is updated.
	//
	// Valid values:
	// - Never: Never refresh the datasource resource when the stack is updated.
	// - Always: Always refresh the datasource resource when the stack is updated.
	// Default is Never.
	RefreshOptions interface{} `field:"optional" json:"refreshOptions" yaml:"refreshOptions"`
	// Property roleName: Filter the results by a specific role name.
	//
	// Supports using * and ?  to fuzzy match.
	RoleName interface{} `field:"optional" json:"roleName" yaml:"roleName"`
}

